HW1/src/main/java/com/hillel/hw1/ValueCalculator.java

```diff
@@ -0,0 +1,188 @@
+ * - Обчислити витрачений час від старту до завершення програми та виведення його в консоль з використанням двох
+ * потоків та з використанням одного потоку
+ * *Зробити шафлінг даних в масиві (перемешати випадковим чином, щоб значення були невідсортовані)
+ * *Додати логіку, яка шукає найбільший елемент у заданому масиві
+ * Зробити це через декілька потоків (кількість потоків обрати самім, наприклад зробити через 2, 5 та 10 потоків
+ * й порівняти я якою кількістю потоків программа найшвидше знайшла максимальне значення)
+ */
+
+
+public class ValueCalculator {
+
+    private final double[] srcArray;
+    private int size;
+    private int halfSize;
+
+    private static long start;
+    private static long end;
+
+
+    public ValueCalculator(int size) {
+        this.size = Math.max(size, 1_000_000);
+        this.halfSize = size / 2;
+        this.srcArray = new double[this.size];
+    }
+
+    public void calculate() throws InterruptedException {
+
+        start = System.currentTimeMillis();
+
+        Arrays.fill(srcArray, 1.0);
+
+        double[] firstPartArray = new double[halfSize];
+        double[] secondPartArray = new double[halfSize];
+
+        System.arraycopy(srcArray, 0, firstPartArray, 0, halfSize);
+        System.arraycopy(srcArray, halfSize, secondPartArray, 0, halfSize);
+
+        Thread thread1 = new Thread(() -> arrayChanger(firstPartArray));
+        Thread thread2 = new Thread(() -> arrayChanger(secondPartArray));
+
+        thread1.start();
+        thread2.start();
+
+        thread1.join();
+        thread2.join();
+
+        System.arraycopy(firstPartArray, 0, srcArray, 0, halfSize);
+
+        System.arraycopy(secondPartArray, 0, srcArray, halfSize, halfSize);
+
+        end = System.currentTimeMillis();
+        System.out.println("Elapsed time with 2 threads: " + (end - start));
+
+        // --------------------------------------------------------------
+
+        start = System.currentTimeMillis();
+        Arrays.fill(srcArray, 1.0);
+        arrayChanger(srcArray);
+        end = System.currentTimeMillis();
+        System.out.println("Elapsed time with 1 thread: " + (end - start));
+
+    }
+
+    public void arrayChanger(double[] arr) {
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = (float) (arr[i] * Math.sin(0.2f + i / 5) * Math.cos(0.2f + i / 5) * Math.cos(0.4f + i / 2));
+        }
+    }
+
+    public void shuffleArray() {
+        Random random = new Random();
+        for (int i = srcArray.length - 1; i > 0; i--) {
+            int randomIndex = random.nextInt(i + 1);
+            double temp = srcArray[randomIndex];
+            srcArray[randomIndex] = srcArray[i];
+            srcArray[i] = temp;
+        }
+    }
+
+    public void findMaxValue() {
+        double maxValue = srcArray[0];
+        start = System.currentTimeMillis();
+        for (double d : srcArray) {
+            if (d > maxValue) {
+                maxValue = d;
+            }
```

Comment:
It's a working solution but to optimise it you can sort this part of array (for example, by quick sort and choose the last one element)